include/mbedtls/cipher.h

```diff
@@ -34,7 +34,7 @@
 #include "mbedtls/platform_util.h"
 
 #if defined(MBEDTLS_GCM_C) || defined(MBEDTLS_CCM_C) || defined(MBEDTLS_CHACHAPOLY_C)
-#define MBEDTLS_CIPHER_MODE_AEAD
+#define MBEDTLS_CIPHER_HAVE_AEAD_LEGACY
```

Comment:
Hmm, I should have noticed earlier, but this is a public header, so `MBEDTLS_CIPHER_MODE_AEAD` was a public macro, so we should probably avoid removing or renaming it. Can we switch back so the original name?

Reply:
Sure we can go back, but my question is: this header is public, but at the same time this is a sort of "internal symbol" that the end user is not supposed to set/rely in his code. Am I wrong?
I renamed it in order to better clarify its composing elements, but perhaps adding the `VIA_LEGACY_OR_USE_PSA` in the new ones might obtain the same result

Reply:
> Sure we can go back, but my question is: this header is public, but at the same time this is a sort of "internal symbol" that the end user is not supposed to set/rely in his code. Am I wrong?

I'd say that's a grey area:
- there's no user documentation either, which public symbols tend to have;
- but there's nothing that says " internal symbol, please don't use in your code";
- and it's in a public header.

(If you ask me, we have too many macros in that grey area. Each macro (function, type, etc) should be unambiguously private or public. But here we are.)

So I wouldn't say renaming it is an absolute no, but if we can avoid it it's better. As you say, I think if we add a distinctive suffix to the new internal macro, hopefully that should be enough to distinguish between the two. (I agree if we didn't have this grey area issue, suffixing both macros would clearly be the right thing to do.)